packages/web-components/fast-element/docs/guide/declaring-templates.md

```diff
@@ -190,4 +190,16 @@ Besides rendering content, attributes, and properties, you'll often want to add
 In both examples above, after your event handler is executed, `preventDefault()` will be called on the event object by default. You can return `true` from your handler to opt-out of this behavior.
 :::
 
-The second example demonstrates an important characteristic of the templating engine: it only supports *unidirectional data flow* (model => view). It does not support *two-way data binding* (model <=> view). As shown above, pushing data from the view back to the model should be handled with explicit events that call into your model's API.
\ No newline at end of file
+The second example demonstrates an important characteristic of the templating engine: it only supports *unidirectional data flow* (model => view). It does not support *two-way data binding* (model <=> view). As shown above, pushing data from the view back to the model should be handled with explicit events that call into your model's API.
+
+## Templating and the element lifecycle
+
+It is during the `connectedCallback` phase of the Custom Element lifecycle that `FASTElement` creates templates and binds the resulting view. The creation of the template only occurs the first time the element is connected, while binding happens every time the element is connected (with unbinding happening during the `disconnectedCallback` for symmetry).
+
+:::note
+In the future we're planning new optimizations that will enable us to safely determine when we do not need to unbind/rebind certain views.
+:::
+
+In most cases, the template that `FASTElement` renders is determined by the `template` property of the Custom Element's configuration. However, you can also implement a method on your Custom Element class named `resolveTemplate()` that returns a template instance. If this method is present, it will be called during `connectedCallback` to obtain the template to use. This allows the element author to dynamically select completely different templates based on the state of the element at the time of connection.
```

Comment:
Is it possible to *conditionally* override the `template` property from this method? For instance, does the controller defer to the `template` property template if the returned value from this is null?

Reply:
If you return `null` from `resolveTemplate()` then it won't render any template. But you can come back later and set `$fastController.template` and that will trigger a render. Due to the timing of the template location now, you could have your `resolveTemplate()` implementation check the state of its attrs and observables and use that to choose one or another template, because that should all be setup by the time the hook gets called.

Does that answer your question?

Reply:
Yep! I am pondering if that capability could be useful but can't think of a solid use-case at the moment

Reply:
I'm fearful of the internal logic getting a bit complicated and hard to reason about. So, I kept it to a straight priority. First check if the property has been set, then check the custom hook, then check the static definition. I think any advanced scenario can be handled by having code set the property as needed. This also allows the property to be surfaced on FoundationElement as an observable and be bound on a per instance basis as well, which would enable per element template or style overrides with very little effort.